lewis/core/statemachine.py

```diff
@@ -294,6 +295,8 @@ def bind_handlers_by_name(self, instance, override=False, prefix=None):
         """
         if prefix is None:
             prefix = {}
+        if not isinstance(prefix, dict) and hasattr(prefix, '__iter__'):
+            prefix = dict(zip(['on_entry', 'in_state', 'on_exit'], prefix))
 
         # Merge prefix defaults with any provided prefixes
         prefix = dict(list(self._prefix.items()) + list(prefix.items()))
```

Comment:
It's probably still better to not modify it, because that will change the behavior when calling the function multiple times. I was just thinking that it might be good to use `dict_strict_update` to catch key-typos in a dict that is passed in.